internal/controllers/rest_controller.go

```diff
@@ -186,3 +193,17 @@ func (rc RESTController) forbiddenResponse(w http.ResponseWriter, err error) {
 	}
 	json.NewEncoder(w).Encode(jsonBody)
 }
+
+// allowlistMiddleware is a middleware that allows all requests that match the allowlist.
+func (rc RESTController) allowlistMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		for _, allow := range rc.configuration.AllowlistURL {
+			pathPrefix := fmt.Sprintf("/%s", strings.TrimPrefix(allow, "/"))
+			if strings.HasPrefix(c.Request.URL.Path, pathPrefix) {
+				rc.logger.Infof("URL %s is allowlisted. Accepted without authorization.", c.Request.URL.Path)
+				c.AbortWithStatus(http.StatusOK)
```

Comment:
AbortWithStatus usually used for failures according to the [doc](https://pkg.go.dev/github.com/gin-gonic/gin#Context.AbortWithStatus), that's an error condition?

That logic is inside the loop, whats happen if you have another loop iteration? Did you will write the status header again?

This logic seems a bit confused.

Reply:
not necessarily, see:

```
// Abort prevents pending handlers from being called. Note that this will not stop the current handler.
// Let's say you have an authorization middleware that validates that the current request is authorized.
// If the authorization fails (ex: the password does not match), call Abort to ensure the remaining handlers
// for this request are not called.
```

it's exactly what i need.
If it's allowlist, I don't want to validate if I have a token or not.

Reply:
let me see again...

Reply:
@renatoaguimaraes look again, please.

```
// Abort prevents pending handlers from being called. Note that this will not stop the current handler.
// Let's say you have an authorization middleware that validates that the current request is authorized.
// If the authorization fails (ex: the password does not match), call Abort to ensure the remaining handlers
// for this request are not called.
```

it's exactly what i need.
If it's allowlist, I don't want to validate if I have a token or not.